discovery-infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -91,8 +99,40 @@ def format_disk(disk_path):
         # Fix for libvirt 6.0.0
         if image_size.isdigit():
             image_size += "G"
-        command = f'qemu-img create -f qcow2 {disk_path} {image_size}'
-        utils.run_command(command, shell=True)
+
+        cls.create_disk(disk_path, image_size)
+
+    def attach_disk(self, node, disk_size):
+        node = self.libvirt_connection.lookupByName(node)
+
+        # <target> dev attribute needs some unique device identifier for the disk (target_dev).
+        # This path will not be the same path used by the guest for this disk.
+        # See https://libvirt.org/formatdomain.html#elementsDisks:~:text=The%20target%20element%20controls%20the%20bus
+        # We generate a random one with a lot of digits so collision is very unlikely
+        # When detaching, we use the alias rather than target_dev,
+        # (unlike virsh attach-disk and detach-disk commands)
+        disk_id = random.randrange(int(1e12))
+        target_dev = f"vdtest{disk_id}"
+        disk_alias = f"ua-testDisk{disk_id}"
+
+        tmp_disk = tempfile.mktemp()
```

Comment:
`mktemp()` is deprecated
https://docs.python.org/3/library/tempfile.html#tempfile.mktemp